menus/pregame_track.py

```diff
@@ -22,9 +22,66 @@ def __init__(self, characters):
         self.progress = progress.Progress()
         self.flags = flags.Flags()
 
+        self.invoke_flags_submenu = {}
         self.characters = characters
         self.mod()
 
+    def get_flags_a_check_src(self, invoke_submenu_addr):
+        # get the ASM for sustain_mod that checks whether we are in the flags menu
+        # and the A button is clicked to launch a submenu.
+        src = [
+            # if on the flags menu, check A button press
+            asm.LDA(0x200, asm.ABS), 
+            asm.CMP(self.flags.MENU_NUMBER, asm.IMM8), # in Flags menu?
+            asm.BNE("HANDLE_SCROLLING"),               # branch if not
+            asm.LDA(0x08, asm.DIR),
+            asm.BIT(0x80, asm.IMM8),        # a pressed?
+            asm.BEQ("HANDLE_SCROLLING"),    # branch if not
+        ]
+
+        for submenu_idx in self.flags.submenus.keys():
+            src += [
+                asm.LDA(0x4b, asm.DIR),         # a = cursor index
+                asm.CMP(submenu_idx, asm.IMM8), # is the cursor index = this submenu?
+                asm.BNE(f"NEXT_SUBMENU_CHECK{submenu_idx}"),    # branch if not
+                asm.TDC(),
+                asm.JSR(0x0eb2, asm.ABS),       # click sound
+                asm.JSR(self.exit_scroll_area, asm.ABS), # save current submenu position
+                asm.JMP(invoke_submenu_addr, asm.ABS), # load the flags submenu
+                f"NEXT_SUBMENU_CHECK{submenu_idx}",
+            ]
+        src += ["HANDLE_SCROLLING"]
+
+        return src
```

Comment:
### Change 2: make `get_flags_a_check_src` fn handle one menu at a time rather than iterating

Unrelated note. The name of this fn confused me - I renamed it, but feel free to change it back
 
```suggestion
    def get_submenu_src(self, submenu_id, invoke_submenu_addr):
        SUBMENU_LABEL = f"SUBMENU_CHECK{submenu_id}"
        SUBMENU_END_LABEL = f"SUBMENU_CHECK{submenu_id}_END"
        HANDLE_SCROLLING_LABEL = f"HANDLE_SCROLLING_{submenu_id}"
        # get the ASM for sustain_mod that checks whether we are in the flags menu
        # and the A button is clicked to launch a submenu.
        src = [
            # if on the flags menu, check A button press
            asm.LDA(0x200, asm.ABS),
            asm.CMP(self.flags.MENU_NUMBER, asm.IMM8), # in Flags menu?
            asm.BNE(HANDLE_SCROLLING_LABEL),               # branch if not
            asm.LDA(0x08, asm.DIR),
            asm.BIT(0x80, asm.IMM8),        # a pressed?
            asm.BEQ(HANDLE_SCROLLING_LABEL),    # branch if not
        ]

        src += [
            SUBMENU_LABEL,
            asm.LDA(0x4b, asm.DIR),         # a = cursor index
            asm.CMP(submenu_id, asm.IMM8), # is the cursor index = this submenu?
            asm.BNE(SUBMENU_END_LABEL),    # branch if not
            asm.TDC(),
            asm.JSR(0x0eb2, asm.ABS),       # click sound
            asm.JSR(self.exit_scroll_area, asm.ABS), # save current submenu position
            asm.JMP(invoke_submenu_addr, asm.ABS), # load the flags submenu
            SUBMENU_END_LABEL,
        ]
        src += [HANDLE_SCROLLING_LABEL]

        return src
```

Reply:
oh. a-check. As in pressed A

Reply:
Thanks, I incorporated this change.